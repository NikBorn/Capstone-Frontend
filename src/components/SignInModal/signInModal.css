.outer-modal{
  position: fixed;
 width: 100%;
 height: 100%;
 top: 0;
 left: 0;
 right: 0;
 bottom: 0;
 margin: auto;
 background-color: rgba(0,0,0, 0.5);
}

.inner-modal{
  position: absolute;
  left: 25%;
  right: 25%;
  top: 10%;
  bottom: 70%;
  margin: auto;
  background: white;
}

.modal-text{
    color: black;
}