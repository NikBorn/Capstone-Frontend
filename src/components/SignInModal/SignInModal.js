import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './signInModal.css';


export default class SignInModal extends Component {

  render() {
    return (
      <div className='outer-modal'>
        <div className='inner-modal'>
          <p className='modal-text'>Please Sign In</p>
          <button onClick={() => this.props.close()}
            className='invoice-input'>
                        Close
          </button>
        </div>
      </div>
    );
  }
}


