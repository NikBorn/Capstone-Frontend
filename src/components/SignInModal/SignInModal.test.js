import { shallow, configure } from 'enzyme';
import SignInModal from './SignInModal';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
/* eslint-disable no-undef */

configure({ adapter: new Adapter() });


describe('SignInModal snapshot', () => {

  it('should always match the snapshot', () => {
      const mkFun = jest.fn();
      const wrapper = shallow(<SignInModal
          close={mkFun} />);

    expect(wrapper).toMatchSnapshot();
  });
});
