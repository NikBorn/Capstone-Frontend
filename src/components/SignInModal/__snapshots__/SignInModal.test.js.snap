// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInModal snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignInModal
    close={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inner-modal"
>
        <p
                className="modal-text"
        >
                Please Sign In
        </p>
        <button
                className="invoice-input"
                onClick={[Function]}
        >
                Close
        </button>
</div>,
      "className": "outer-modal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="modal-text"
>
            Please Sign In
</p>,
          <button
            className="invoice-input"
            onClick={[Function]}
>
            Close
</button>,
        ],
        "className": "inner-modal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please Sign In",
            "className": "modal-text",
          },
          "ref": null,
          "rendered": "Please Sign In",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Close",
            "className": "invoice-input",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Close",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner-modal"
>
          <p
                    className="modal-text"
          >
                    Please Sign In
          </p>
          <button
                    className="invoice-input"
                    onClick={[Function]}
          >
                    Close
          </button>
</div>,
        "className": "outer-modal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="modal-text"
>
              Please Sign In
</p>,
            <button
              className="invoice-input"
              onClick={[Function]}
>
              Close
</button>,
          ],
          "className": "inner-modal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Please Sign In",
              "className": "modal-text",
            },
            "ref": null,
            "rendered": "Please Sign In",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Close",
              "className": "invoice-input",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Close",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
